update

```diff
@@ -33,13 +34,22 @@ dirs.forEach(function (dir) {
       return fs.readFileSync(path.join(__dirname, dir, file))
     }
 
+    function readBuiltinVersion (file) {
+      // NOTE: should be using this, but it leaks memory
+      // return solc(require(path.join(__dirname, dir, file))).version()
+      const filename = path.join(__dirname, dir, file)
+      return cp.execSync(`/usr/bin/env node -e "var solc = require('${filename}'); console.log(solc.cwrap(('_solidity_version' in solc) ? 'solidity_version' : 'version', 'string', [])())"`).toString().trim()
+    }
+
     // ascending list (oldest version first)
     const parsedList = files
       .map(function (file) { return file.match(/^soljson-v([0-9.]*)(-([^+]*))?(\+(.*))?.js$/) })
       .filter(function (version) { return version })
       .map(function (pars) { return { path: pars[0], version: pars[1], prerelease: pars[3], build: pars[5] } })
       .map(function (pars) {
+        console.log('Processing ' + pars.path)
         const fileContent = readFile(pars.path)
+        pars.reportedVersion = readBuiltinVersion(pars.path)
```

Comment:
One problem with adding it now is that the repo contains binaries for multiple platforms and you won't be able to get versions for all of them in one run. We'd need separate runs for different platforms and then we would have to combine gathered info.